clmm/theory/parent_class.py

```diff
@@ -735,31 +1086,133 @@ def eval_magnification(self, r_proj, z_cl, z_src, verbose=False):
             The projected radial positions in :math:`M\!pc`.
         z_cl : float
             Galaxy cluster redshift
-        z_src : array_like, float
-            Background source galaxy redshift(s)
+        z_src : array_like, float, function
+            Information on the background source galaxy redshift(s). Value required depends on
+            `z_src_info` (see below).
+        z_src_info : str, optional
+            Type of redshift information provided, it describes z_src.
```

Comment:
Same docstring comments as before for the whole new redshift-related docstrings.